### DISCUSSION TITLE
How to speed up binary transfer?

### DISCUSSION THREAD

**Discussion by adamplonka** [microsoft/vs-streamjsonrpc#816] [Q&A] [answered]
I'm trying to send large binary file using MultiplexingStream over TcpSocket. I tried to send it directly using sockets (on loopback interface) and I'm able to transfer over 10GB/s.
When I try to do the same using MultiplexingStream and return Stream or create separate channel I could get up to only 80MB/s.

I'm using `LengthHeaderMessageHandler` and `MessagePackFormatter` but it shouldn't matter as I send binary over separate channel. I tried to increase the buffer size but with no luck. Does the MultiplexingStream add so much overhead that it's not possible to get it any faster?

**Comment by AArnott** [comment, marked as answer]
This isn't really a question for StreamJsonRpc, since as you say you're on a separate channel from RPC when you're transferring the data. Can you post again over at https://github.com/AArnott/Nerdbank.Streams/discussions?

**Reply by adamplonka** [reply]
Sure, posted here https://github.com/AArnott/Nerdbank.Streams/discussions/503
